File: buildSrc/src/main/groovy/com/engineer/plugin/utils/BeautyLog.java
package com.engineer.plugin.utils;

import org.gradle.api.logging.Logger;

/**
 * @author zhuyongging @ Zhihu Inc.
 * @since 11-19-2019
 */
public class BeautyLog {
    public static void log(String tag, boolean isStart, Logger logger) {
        if (isStart) {
            System.out.println();
            String msg =
                    "=================================== " + tag + " ===================================>>>>";
            logger.error(msg);
            System.out.println();
        } else {
            System.out.println();
            String msg =
                    "<<<<=================================== " + tag + " ===================================";
            logger.error(msg);
            System.out.println();
        }
    }
}

File: buildSrc/src/main/kotlin/com/engineer/plugin/utils/BeautyLog.kt
package com.engineer.plugin.utils

import org.gradle.api.logging.Logger

/**
 * @author rookie
 * @since 11-19-2019
 */
object BeautyLog {
    fun log(
        tag: String,
        isStart: Boolean,
        logger: Logger
    ) {
        if (isStart) {
            val msg =
                "\n=========================================== $tag ======================================== >>>>\n"
            logger.error(msg)
        } else {
            val msg =
                "\n<<<< ======================================= $tag ===========================================\n"
            logger.error(msg)
        }
    }

    fun log(
        tag: String,
        isStart: Boolean
    ) {
        if (isStart) {
            val msg =
                "\n=========================================== $tag ======================================== >>>>\n"
            System.err.print(msg)
        } else {
            val msg =
                "\n<<<< ======================================= $tag ===========================================\n"
            System.err.print(msg)
        }
    }

    fun dog(info: String, logger: Logger) {
        logger.error("\n=================================================================================== >>>>\n")
        logger.error(info)
        logger.error("\n=================================================================================== >>>>\n")
    }

    fun dog(info: String) {
        System.err.print(info)
    }
}